website/content/labs/lab6.md

```diff
@@ -0,0 +1,176 @@
+
+* [5 points] Make a change to the `hello.c` file that will make it not build. Commit and push this change to the repository. This will result in a failed pipeline in the “Pipelines” page; copy-paste the URL of the failed pipeline into the `tasks.txt` file. Before moving on to the next task, undo the change and verify that pushing the code results in a “passed” pipeline.
+
+* [15 points] Add a job called `libgeometry:build` to build the libgeometry library contained in `lab6/libgeometry`. Push your updated `.gitlab-ci.yml` file, and copy-paste the URL of the resulting “passed” pipeline into the `tasks.txt` file. Notice how this pipeline now has two jobs in it.
+
+Note: Before committing your `.gitlab-ci.yml` file, you can check that the format of the file is correct by going to https://mit.cs.uchicago.edu/ci/lint. This will allow you to copy-paste your `.gitlab-ci.yml` file and check it for syntax errors.
+
+
+
+# Task 2: Adding tests
+
+As we mentioned previously, CI typically involves running tests every time we push changes to the repository. This means that, besides building our code, we also want tests to run automatically. For the simple hello program, we’ve included a very simple test: we simply run the program (using the `tests` target in the Makefile). While this doesn’t guarantee that the program prints the correct output, it will at least verify that the program runs without errors.
+
+So, let’s add a job to the `.gitlab-ci.yml` file to run the tests:
+
+    hello:test:
+        stage: test
+        script:
+          - make -C lab6/hello clean tests
+        only: 
+          - master
+
+Notice how the stage is `test`. This means that GitLab will not run this job unless all the jobs in the `build` stage succeed.
+
+Now, do the following:
+
+* [5 points] Commit the updated `.gitlab-ci.yml` file, and copy-paste the URL of the resulting “passed” pipeline to the `tasks.txt` file. Notice how GitLab now shows a `hello:test` job that depends on the build jobs. If you click on the `hello:test` job and look at its terminal output, you will see the output of running `hello`.
+* [10 points] Edit `hello.c` to force it to crash (e.g., dereference a NULL pointer). This will still allow `hello.c` to compile, but will make the tests fail. Commit and push this change, and copy-paste the URL of the resulting “failed” pipeline to the `tasks.txt` file. Notice how the build jobs ran successfully, but the pipeline failed because one of the jobs (the `hello:test` job) failed. Before moving on to the next test, undo this change and make sure it results in a “passed” pipeline.
```

Comment:
URLs again. We should also have them identify in their own words where the failure showed up in the pipeline output.